models.py

```diff
@@ -26,22 +28,129 @@ def get_attr_by_path(self, path: str, sep: str = "__"):
 
         return current_level
 
-
     def get_indicators(self) -> dict:
         indicators = {}
         for indicator in self.indicators:
             value = self.get_attr_by_path(indicator["id"])
             cmp = indicator["not"] if isinstance(indicator["not"], list) else [indicator["not"]]
             indicators[f"has_{indicator['id']}"] = all(value != c for c in cmp)
+
         return indicators
 
+    def get_prefix_or_fallback_from(self, url: str, message: str) -> str:
+        m = re.match("^(.*/)[^/]+$", url)
+
+        if m:
+            return m.group(1)
+
+        return message
+
+    def compute_prefix_harvest_remote_id(self) -> str:
+        missing_prefix_message = 'Préfixe manquant'
+
+        try:
+            if (self.payload is None):
+                raise KeyError
+
+            harvest = self.payload['harvest']
+
+            if (harvest is None):
+                raise KeyError
+
+            remote_id = harvest['remote_id']
+
+            if (remote_id is None):
+                raise KeyError
+        except KeyError:
+            return missing_prefix_message
+
+        return self.get_prefix_or_fallback_from(
+            remote_id,
+            missing_prefix_message
+        )
+
+    def compute_prefix_harvest_remote_url(self) -> str:
+        missing_prefix_message = 'Préfixe manquant'
+
+        try:
+            if (self.payload is None):
+                raise KeyError
+
+            harvest = self.payload['harvest']
+
+            if (harvest is None):
+                raise KeyError
+
+            remote_url = harvest['remote_url']
+
+            if (remote_url is None):
+                raise KeyError
+        except KeyError:
+            return missing_prefix_message
+
+        return self.get_prefix_or_fallback_from(
+            remote_url,
+            missing_prefix_message
+        )
+
+    def get_url_data_gouv(self) -> str:
+        url = 'https://{prefix}.data.gouv.fr/fr/datasets/'.format(
+            prefix=self.prefix
+        )
+        id = self.payload['id']
+
+        return "<a href=\"{url}{id}\" target=\"_blank\">{id}</a>".format(
+            url=url, id=id
+        )
+
+    def get_consistent_dates(self) -> bool:
+        try:
+            created_at = self.payload['created_at']
+        except KeyError:
+            return False
+
+        try:
+            modified_at = self.payload['last_modified']
+        except KeyError:
+            return True
+
+        return modified_at >= created_at
+
+    def get_consistent_temporal_coverage(self) -> bool:
+        temporal_coverage = self.payload['temporal_coverage']
+
+        if temporal_coverage is None:
+            return False
```

Comment:
```suggestion
            return True
```

Pas d'incohérence si pas de métadonnée déclarée.